### PR TITLE
Cherrypicks to prepare for 2.1.5

### PR DESCRIPTION
**Pull request by mattcary**
/kind bug

**What this PR does / why we need it**:
Cherry-pick #478, and #413 in preparation for 2.1.5 release.

```release-note
None
```

### CODE REVIEW

**Comment by mattcary** [comment]
/hold

They didn't apply as cleanly as I thought it did.

**Comment by mattcary** [comment]
/hold cancel

The csi-lib-utils merge was indeed a bridge too far.

File: CHANGELOG-2.1.md [hunk 1]
```diff
@@ -1,3 +1,21 @@
+# Release notes for v2.1.5 (Changelog since v2.1.4)
+
+## Changes by Kind
+
+### Bug or Regression
+
+- Cherry pick of #413: Bug fix to allow creation of snapshot content if pvc finalizer exists, even if pvc is marked for deletion.
+- Cherry pick of #478: Call dynamic VS content creation unconditionally.
```
Comment (xing-yang) [inline CHANGELOG-2.1.md:8]: Did you cherry pick #478 in this PR?  I don't see it.

Comment (saikat-royc) [reply CHANGELOG-2.1.md:8]: the wording is a bit confusing. But #478 itself is the actual cherrypick to release-1.2

Comment (xing-yang) [reply CHANGELOG-2.1.md:8]: Can you reword these two lines following this changelog?
https://github.com/kubernetes-csi/external-snapshotter/blob/v3.0.3/CHANGELOG/CHANGELOG-3.0.md#bug-or-regression

Comment (mattcary) [reply CHANGELOG-2.1.md]: Sorry for the confusion, done.

Note that #413 is cherrypicked in this PR (#488), I hope that's ok.

**Pull request merged**